Source/Scene/Camera.js

```diff
@@ -361,6 +362,17 @@ Camera.prototype._updateCameraChanged = function () {
 
   var percentageChanged = camera.percentageChanged;
 
+  if (!defined(camera._changedHeading)) {
+    camera._changedHeading = camera.heading;
+  }
+
+  var headingChanged = camera._changedHeading === camera.heading ? false : true;
+
+  if (headingChanged) {
+    camera._changed.raiseEvent("Camera Heading Changed");
+    camera._changedHeading = camera.heading;
+  }
+
```

Comment:
Avoid calling `camera.heading` multiple times since it's a pretty expensive function. You should be able to call it just once and reuse the result.

Reply:
Nice catch. I did not realize that `camera.heading` was expensive but after looking through the source code that makes sense.